Sources/SwiftFusion/Inference/ChordalInitialization.swift

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+import TensorFlow
+
+/// Type shorthands used in the relaxed pose graph
+public typealias Jacobian9x9_1 = Array9<Tuple1<Vector9>>
+public typealias Jacobian9x9_2 = Array9<Tuple2<Vector9, Vector9>>
+public typealias JacobianFactor9x9_1 = JacobianFactor<Jacobian9x9_1, Vector9>
+public typealias JacobianFactor9x9_2 = JacobianFactor<Jacobian9x9_2, Vector9>
+
+/// Chordal Initialization for Pose3s
+public struct ChordalInitialization {
+  public var anchorId: TypedID<Pose3, Int>
+  
+  public init() {
+    anchorId = TypedID<Pose3, Int>(0)
+  }
+  
+  /// Extract a subgraph of the original graph with only Pose3s.
+  public func buildPose3graph(graph: FactorGraph) -> FactorGraph {
+    var pose3Graph = FactorGraph()
+
+    for factor in graph.factors(type: BetweenFactor3.self) {
+        pose3Graph.store(factor)
+    }
+    
+    for factor in graph.factors(type: PriorFactor3.self) {
+      pose3Graph.store(BetweenFactor3(anchorId, factor.edges.head, factor.prior))
+    }
+    
+    return pose3Graph
+  }
+  
+  /// Construct the orientation graph (unconstrained version of the between in Frobenius norm).
+  public func buildLinearOrientationGraph(
+    g: FactorGraph,
+    v: VariableAssignments,
+    v_linear: VariableAssignments,
+    associations: Dictionary<Int, TypedID<Vector9, Int>>
+  ) -> GaussianFactorGraph {
```

Comment:
Why do you build a Gaussian factor graph? Why not a nonlinear one and let autodiff do its work?

Reply:
I considered all options before implementing, and this is the fastest way to get it working. I can try make a NL version and compare against this version.

Reply:
Hmm, seems counter-intuitive to me. You're manually doing all derivatives in system where all this is done for you, and this is somehow faster? :-)

Reply:
It's a bad design choice at that time, I just wanted to avoid touching the protocol conformances but turned out to be much too complicated than the autodiffed version. I'll push the autodiffed version in ~10mins.

Reply:
done